sklearn/decomposition/_kernel_pca.py

```diff
@@ -56,10 +63,31 @@ class KernelPCA(TransformerMixin, BaseEstimator):
         Learn the inverse transform for non-precomputed kernels.
         (i.e. learn to find the pre-image of a point)
 
-    eigen_solver : string ['auto'|'dense'|'arpack'], default='auto'
-        Select eigensolver to use. If n_components is much less than
-        the number of training samples, arpack may be more efficient
-        than the dense eigensolver.
+    eigen_solver : string {'auto', 'dense', 'arpack', 'randomized'}
+        Select eigensolver to use (default='auto'). If `n_components` is much
+        less than the number of training samples, randomized (or arpack to a
+        smaller extend) may be more efficient than the dense eigensolver.
+        Randomized SVD is performed according to the method of Halko et al.
+
+        auto :
+            the solver is selected by a default policy based on `K.shape[0]`
+            (the number of training samples) and `n_components`:
+            if the number of components to extract is lower than 80% of
+            `K.shape[0]`, then the more efficient 'randomized'
+            method is enabled. Otherwise the exact full eigenvalue
+            decomposition is computed and optionally truncated afterwards.
+        dense :
+            run exact full eigenvalue decomposition calling the standard
+            LAPACK solver via `scipy.linalg.eigh`, and select the components
+            by postprocessing
```

Comment:
what do you mean by postprocessing?

Reply:
I just copied the comment from https://github.com/scikit-learn/scikit-learn/blob/95d4f0841/sklearn/decomposition/_pca.py#L171, did not try to modify it.

Reply:
> did not try to modify it as it was already considered "valid" by sklearn team

lol you can't imagine the number of things in the codebase that bothers us

But OK let's leave it as is I guess